src/lamp_py/performance_manager/l1_rt_trips.py

```diff
@@ -326,12 +329,145 @@ def update_start_times(db_manager: DatabaseManager) -> None:
         )
 
 
+def update_branch_trunk_route_id(db_manager: DatabaseManager) -> None:
+    """
+    update `branch_route_id` and `trunk_route_id` fields in trips table
+    """
+    distinct_t_trips = (
+        sa.select(TempEventCompare.service_date, TempEventCompare.pm_trip_id)
+        .distinct()
+        .subquery("distinct_trips")
+    )
+
+    distinct_trips = (
+        sa.select(
+            VehicleTrips.pm_trip_id,
+            VehicleTrips.route_id,
+            VehicleTrips.branch_route_id,
+            VehicleTrips.trunk_route_id,
+        )
+        .select_from(VehicleTrips)
+        .join(
+            distinct_t_trips,
+            sa.and_(
+                distinct_t_trips.c.service_date == VehicleTrips.service_date,
+                distinct_t_trips.c.pm_trip_id == VehicleTrips.pm_trip_id,
+            ),
+        )
+    )
+
+    distinct_red_trips = (
+        sa.select(TempEventCompare.service_date, TempEventCompare.pm_trip_id)
+        .distinct()
+        .where(TempEventCompare.route_id == "Red")
+        .subquery("distinct_trips")
+    )
+
+    red_events = (
+        sa.select(
+            VehicleEvents.pm_trip_id,
+            VehicleEvents.stop_id,
+        )
+        .select_from(VehicleEvents)
+        .join(
+            distinct_red_trips,
+            sa.and_(
+                distinct_red_trips.c.service_date == VehicleEvents.service_date,
+                distinct_red_trips.c.pm_trip_id == VehicleEvents.pm_trip_id,
+            ),
+        )
+    )
+
+    process_logger = ProcessLogger("l1_trips.update_branch_trunk_route_id")
+    process_logger.log_start()
+
+    trips_df = db_manager.select_as_dataframe(distinct_trips)
+    red_events_df = db_manager.select_as_dataframe(red_events)
```

Comment:
pull two dataframes from DB:

trips_df:  all trips loaded during this event loop (with comparison to TempEventCompare table)
red_events_df: `pm_trip_id` and `stop_id` for all trips found in TempEventCompare table (for redline branch_route_id assignment)